server/console_storage.go

```diff
@@ -111,17 +123,83 @@ func (s *ConsoleServer) ListStorage(ctx context.Context, in *console.ListStorage
 		}
 		userID = &uid
 	}
-
 	var query string
+
+	collections := make([]string, 0)
+	query = "SELECT DISTINCT collection FROM storage"
+	rows, err := s.db.QueryContext(ctx, query)
+	if err != nil {
+		s.logger.Error("Error querying storage collections.", zap.Any("in", in), zap.Error(err))
+		return nil, status.Error(codes.Internal, "An error occurred while trying to list storage collections.")
+	}
+	for rows.Next() {
+		var dbCollection string
+		if err := rows.Scan(&dbCollection); err != nil {
+			_ = rows.Close()
+			s.logger.Error("Error scanning storage collections.", zap.Any("in", in), zap.Error(err))
+			return nil, status.Error(codes.Internal, "An error occurred while trying to list storage collecctions.")
+		}
+		collections = append(collections, dbCollection)
+	}
+
+	args := make([]string, 0)
 	params := make([]interface{}, 0, 1)
-	if userID == nil {
-		query = "SELECT collection, key, user_id, value, version, read, write, create_time, update_time FROM storage LIMIT 50"
-	} else {
-		query = "SELECT collection, key, user_id, value, version, read, write, create_time, update_time FROM storage WHERE user_id = $1"
+	query = ""
+	if userID != nil {
+		args = append(args, "user_id =")
 		params = append(params, *userID)
 	}
+	if in.Key != "" {
+		args = append(args, "key =")
+		params = append(params, in.Key)
+	}
+	if in.Collection != "" {
+		args = append(args, "collection =")
+		params = append(params, in.Collection)
+	}
+	for i, arg := range args {
+		if query == "" {
+			query += " WHERE "
+		} else {
+			query += " AND "
+		}
+		query += fmt.Sprintf("%s $%d", arg, i+1)
+	}
 
-	rows, err := s.db.QueryContext(ctx, query, params...)
+	sc := &consoleStorageCursor{}
+	if in.Cursor != "" {
+		cb, err := base64.RawURLEncoding.DecodeString(in.Cursor)
```

Comment:
We usually use gob encoding for cursor - https://github.com/heroiclabs/nakama/blob/master/server/core_storage.go#L384

I suggest we stick to gob here as well.